### ISSUE TITLE
Add Option for Number Row Key Borders

### ISSUE THREAD

**Issue by KaKi87** [futo-org/android-keyboard#627]
### Use Cases

When number row and key borders are both enabled, number row keys used to have border (in stable), but not anymore (in nightly). 

### Proposal

Add a specific "number row key borders" option below the general "key borders" option, or bring back number row key borders into the general option as it was in stable.

**Comment by FireNinja7365** [comment]
Also the long press keys don't work for the number row unless going to the symbols page

**Comment by KaKi87** [comment]
Why ?

**Comment by trekshcool** [comment]
In addition to the no border number row, I also noticed that long press no longer works on the number row.

The removing borders for numbers makes no sense to me, neither does removing the long press feature from the number row outside of the symbols plane.

**Comment by chileverde** [comment]
I don't understand why this was changed. Especially with so many issues that actually affect the functioning of the keyboard, why change the design so the number keys don't look like keys and also take away the long-press features, unless people switch to symbols keyboard?

I'll stick with v0.1.23.2 for now. Can't think of any reason to upgrade.

**Comment by dean-futo** [comment]
@abb128 in after further considering the points @trekshcool and @chileverde are bringing up, I think it'd be a good idea to make borders toggle-able and also enable long-press for the number row. Whether the number row is located in the symbols menu or on the main keyboard screen, it seems that that the functionality of each should be the same.

**Comment by FireNinja7365** [comment]
Temporary layout for if you want number row borders (unfinished but still works):

[Number Row Fix.txt](https://github.com/user-attachments/files/18480834/Number.Row.Fix.txt)

**Comment by KaKi87** [comment]
Hello,
Any news on this ?
Thanks

**Comment by trekshcool** [comment]
This was not fixed in the latest update please have look and give this issue some priority. It looks like a relativity easy fix.

**Comment by KaKi87** [comment]
@Zvonimir-FUTO Any news on this please ? 🙏

**Comment by Zvonimir-FUTO** [comment]
@KaKi87 please don't @ us. This won't do anything except your comments being marked as spam.

Once the implementation has been developed, issue will be labeled as "Solved" meaning the work has been done but not yet released to the public.

When the new version which includes the implementation has been released, we will close the issue as resolved.

**Comment by KaKi87** [comment]
Okay, but I'm requesting progress information.

This is effectively a regression, which is not rare on this project, [even when we warn while the issue is only present in nightly, yet it ends up released in stable](https://github.com/futo-org/android-keyboard/issues/541#issuecomment-2453976003).

**Comment by abb128** [comment]
It's a design decision that doesn't affect the functionality in any way, that's why it hasn't been worked on. The reasoning for this design is that it better separates numbers from the letters and reduces overall visual clutter when the number row is enabled. The arrow row follows the same principle.

This was mentioned before but you can actually already add a custom layout to get number row borders, in fact here's a cleaner one you can use to get number row key borders and identical height:
```yaml
name: QWERTY
rows:
  - numbers: 1 2 3 4 5 6 7 8 9 0
    rowHeight: 1.0
    attributes:
      style: Normal
  - letters: q w e r t y u i o p
  - letters: a s d f g h j k l
  - letters: z x c v b n m
```

Personally I don't think there would be much value in adding another option just for this, there are already a lot of options and we're trying to keep new ones to a minimum as a lot of users already get overwhelmed and fail to find options. So I hope you can understand my decision that this will probably not be added.

**Comment by trekshcool** [comment]
It absolutely has a functional impact, the hold press on the number row does not work and they are harder to separate without the borders.

Same for shrinking the pop up which makes it less readable for those with thick fingers. All these were downgrades and none of them have been fixed. 

These are not new features they used to be in the keyboard and then were removed or not re-added when the overhaul happened.

**Comment by KaKi87** [comment]
> you can actually already add a custom layout to get number row borders

That's more work than a simple checkbox and it's dismissing the fact that some of use use different keyboard layouts, like French AZERTY for me.

**Comment by FireNinja7365** [comment]
i agree, the key borders on the numbers  just looks way better.. it really isn't that hard to organize the gui a little to fix said clutter, then adding one more checkbox wouldn't be a big deal.

**Comment by FireNinja7365** [comment]
probably can close this now that it's in the nightly build 🎉

**Comment by Zvonimir-FUTO** [comment]
> probably can close this now that it's in the nightly build 🎉

We prefer to keep the issue opened until it's released in stable so users who use stable can find this issue and not open new feature requests.

For those wondering where the option is, it's inside **Keyboard & Typing** > click on **Show Number Row** (no just enable) > at the bottom you will see **Number row style** and you can find Compact and Key styles.

**Comment by KaKi87** [comment]
> We prefer to keep the issue opened until it's released in stable

Thank you for that.

There are so many developers who close issues as soon as they commit, even when they don't have nightly builds, leaving users unable to be notified when an update actually containing the commit is released.

**Comment by NicholasFlamy** [comment]
> We prefer to keep the issue opened until it's released in stable so users who use stable can find this issue and not open new feature requests.

This is greatly appreciated, kudos for doing this!
